app/Community/Actions/AddToMessageThreadAction.php

```diff
@@ -12,11 +12,16 @@
 
 class AddToMessageThreadAction
 {
-    public function execute(MessageThread $thread, User $userFrom, string $body): void
-    {
+    public function execute(
+        MessageThread $thread,
+        User $userFrom,
+        User $trueSenderUser,
+        string $body
+    ): void {
         $message = new Message([
             'thread_id' => $thread->id,
-            'author_id' => $userFrom->ID,
+            'author_id' => $userFrom->id,
+            'sent_by_id' => $trueSenderUser->id,
```

Comment:
Rather than require the upgrade step, we could assume that null means same as author (like we do for achievement unlocks).

Reply:
I think this be displayed when using the share account (if different than author_id).

Reply:
Added in https://github.com/RetroAchievements/RAWeb/pull/3435/commits/f7c8b842aa6d9bf4f0db1dd9d425d9ae064965aa.

![Screenshot 2025-04-27 at 9 42 58 AM](https://github.com/user-attachments/assets/d8710817-fc27-4df9-af5e-a07fe6218364)

Special care has been taken to not accidentally leak this value to the other user. It is only added to props if the user is authorized to see it.

I've also tweaked the logic for writing `sent_by_id` per your suggestion.